File: ListenToWikipedia/Audio/PentatonicMode.swift
import Foundation

enum PentatonicMode: String, CaseIterable, Identifiable {
    case majorPentatonic = "Major Pentatonic"
    case egyptian = "Egyptian"
    case bluesMinor = "Blues Minor"
    case bluesMajor = "Blues Major"
    case minorPentatonic = "Minor Pentatonic"

    var id: String { rawValue }

    var intervals: [Int] {
        switch self {
        case .majorPentatonic: return [2, 2, 3, 2, 3]
        case .egyptian:        return [2, 3, 2, 3, 2]
        case .bluesMinor:      return [3, 2, 3, 2, 2]
        case .bluesMajor:      return [2, 3, 2, 2, 3]
        case .minorPentatonic: return [3, 2, 2, 3, 2]
        }
    }
}

File: android/app/src/main/java/me/bryanoltman/listentowikipedia/audio/PentatonicMode.kt
package me.bryanoltman.listentowikipedia.audio

enum class PentatonicMode(val displayName: String, val intervals: List<Int>) {
    MAJOR_PENTATONIC("Major Pentatonic", listOf(2, 2, 3, 2, 3)),
    EGYPTIAN("Egyptian", listOf(2, 3, 2, 3, 2)),
    BLUES_MINOR("Blues Minor", listOf(3, 2, 3, 2, 2)),
    BLUES_MAJOR("Blues Major", listOf(2, 3, 2, 2, 3)),
    MINOR_PENTATONIC("Minor Pentatonic", listOf(3, 2, 2, 3, 2))
}
